lms/djangoapps/discussion/rest_api/utils.py

```diff
@@ -231,15 +232,67 @@ def create_topics_v3_structure(blocks, topics):
             value['courseware'] = True
             courseware_topics.append(value)
             value['children'] = create_discussion_children_from_ids(
-                value['children'],
+                value.get('children', []),
                 blocks,
                 topics,
             )
             subsections = value.get('children')
             for subsection in subsections:
                 subsection['children'] = create_discussion_children_from_ids(
-                    subsection['children'],
+                    subsection.get('children', []),
                     blocks,
                     topics,
                 )
-    return non_courseware_topics + courseware_topics
+
+    structured_topics = non_courseware_topics + courseware_topics
+    topic_ids = get_topic_ids_from_topics(topics)
+
+    # Remove all topic ids that are contained in the structured topics
+    for chapter in structured_topics:
+        for sequential in chapter.get('children', []):
+            for item in sequential['children']:
+                topic_ids.remove(item['id'])
+
+    archived_topics = {
+        'id': "archived",
+        'children': get_archived_topics(topic_ids, topics)
+    }
+    if archived_topics['children']:
+        structured_topics.append(archived_topics)
+
+    return structured_topics
+
+
+def get_topic_ids_from_topics(topics: List[Dict[str, str]]) -> List[str]:
+    """
+    This function takes a list of topics and returns a list of the topic ids.
+
+    Args:
+    - topics (List[Dict[str, str]]): A list of topic dictionaries. Each dictionary should have an 'id' field.
+
+    Returns:
+    - A list of topic ids, extracted from the input list of topics.
+    """
+    return [topic['id'] for topic in topics]
+
+
+def get_archived_topics(filtered_topic_ids: List[str], topics: List[Dict[str, str]]) -> List[Dict[str, str]]:
+    """
+    This function takes a list of topic ids and a list of topics, and returns the list of archived topics.
+
+    A topic is considered archived if it has a non-null `usage_key` field.
+
+    Args:
+    - filtered_topic_ids (List[str]): A list of topic ids to filter on.
+    - topics (List[Dict[str, str]]): A list of topic dictionaries.
+    - Each dictionary should have a 'id' and a 'usage_key' field.
+
+    Returns:
+    - A list of archived topic dictionaries, with the same format as the input topics.
+    """
+    archived_topics = []
+    for topic_id in filtered_topic_ids:
+        for topic in topics:
+            if topic['id'] == topic_id and topic['usage_key'] is not None:
```

Comment:
A topic is archived if usage key is not None? I thought it should be the other way round

Reply:
This is only to filter out non-courseware topics i.e  General, as these topics do not contain children.